sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jdbc/JdbcConnectionString.java

```diff
@@ -0,0 +1,136 @@
+    public static final String INVALID_PROPERTY_PAIR_FORMAT = "Connection string has invalid key value pair: %s";
+    private static final String TOKEN_VALUE_SEPARATOR = "=";
+    private final String jdbcURL;
+    private final Map<String, String> properties = new HashMap<>();
+    private DatabaseType databaseType = null;
+
+    private JdbcConnectionString(String jdbcURL) {
+        this.jdbcURL = jdbcURL;
+    }
+
+    private void resolveSegments() {
+        if (!StringUtils.hasText(this.jdbcURL)) {
+            LOGGER.warn("'connectionString' doesn't have text.");
+            throw new IllegalArgumentException(String.format(INVALID_CONNECTION_STRING_FORMAT, this.jdbcURL));
+        }
+
+        Optional<DatabaseType> optionalDatabaseType = Arrays.stream(DatabaseType.values())
+                                                            .filter(databaseType -> this.jdbcURL.startsWith(databaseType.getSchema()))
+                                                            .findAny();
+        this.databaseType = optionalDatabaseType.orElseThrow(() -> new AzureUnsupportedDatabaseTypeException(String.format(UNSUPPORTED_DATABASE_TYPE_STRING_FORMAT, this.jdbcURL)));
+
+        int pathQueryDelimiterIndex = this.jdbcURL.indexOf(this.databaseType.getPathQueryDelimiter());
+
+        if (pathQueryDelimiterIndex < 0) {
+            return;
+        }
+
+        String hostInfo = this.jdbcURL.substring(databaseType.getSchema().length() + 3, pathQueryDelimiterIndex);
+        String[] hostInfoArray = hostInfo.split(":");
+        if (hostInfoArray.length == 2) {
+            this.properties.put("servername", hostInfoArray[0]);
+            this.properties.put("port", hostInfoArray[1]);
+        } else {
+            this.properties.put("port", hostInfo);
+        }
+
+        String properties = this.jdbcURL.substring(pathQueryDelimiterIndex + 1);
+
+        final String[] tokenValuePairs = properties.split(this.databaseType.getQueryDelimiter());
+
+        for (String tokenValuePair : tokenValuePairs) {
+            final String[] pair = tokenValuePair.split(TOKEN_VALUE_SEPARATOR, 2);
+            String key = pair[0];
+            if (!StringUtils.hasText(pair[0])) {
+                throw new IllegalArgumentException(String.format(INVALID_PROPERTY_PAIR_FORMAT, tokenValuePair));
+            }
+            if (pair.length < 2) {
+                this.properties.put(key, NONE_VALUE);
+            } else {
+                this.properties.put(key, pair[1]);
+            }
+        }
+    }
+
+    public String enhanceConnectionString(Map<String, String> enhancedProperties) {
+        if (enhancedProperties == null || enhancedProperties.isEmpty()) {
+            return this.jdbcURL;
+        }
+        LOGGER.debug("Trying to enhance url for {}", databaseType);
+
+        StringBuilder builder = new StringBuilder(this.jdbcURL);
+
+        if (!this.hasProperties()) {
```

Comment:
this logic is wired.

why if  the object doesn't `hasProperties` then use `QueryDelimiter` or `PathQueryDelimiter`

Reply:
Yes, it's a little wired here,
`QueryDelimiter` may be different from `PathQueryDelimiter`,
- If jdbcurl has no properties like ` jdbc:postgresql://postgre:5432/test`, before adding enhanced properties, we need to append  a `?` at the end.
- If the jdbcurl has properties like ` jdbc:postgresql://postgre:5432/test?azure.clientId=1234`, we need to append  a `&` at the end.

```
 jdbc:postgresql://postgre:5432/test?azure.clientId=1234
 jdbc:postgresql://postgre:5432/test?azure.clientId=1234&azure.clientSecret=****
 jdbc:postgresql://postgre:5432/test
```

Reply:
so make the code more readable